clang 9.0 report warning: unused type alias 'size_type' [-Wunused-local-typedef]

When I use this project under clang 9.0, the compiler complaints below warning messages:
```cpp
nlohmann/json/single_include/nlohmann/json.hpp:11437:15: warning: unused type alias 'size_type' [-Wunused-local-typedef]
        using size_type = typename BasicJsonType::size_type;
              ^
nlohmann/json/single_include/nlohmann/json.hpp:11510:15: warning: unused type alias 'size_type' [-Wunused-local-typedef]
        using size_type = typename BasicJsonType::size_type;
              ^
nlohmann/json/single_include/nlohmann/json.hpp:11570:15: warning: unused type alias 'size_type' [-Wunused-local-typedef]
        using size_type = typename BasicJsonType::size_type;
              ^
nlohmann/json/single_include/nlohmann/json.hpp:11620:15: warning: unused type alias 'size_type' [-Wunused-local-typedef]
        using size_type = typename BasicJsonType::size_type;
              ^
nlohmann/json/single_include/nlohmann/json.hpp:11663:15: warning: unused type alias 'size_type' [-Wunused-local-typedef]
        using size_type = typename BasicJsonType::size_type;
              ^
nlohmann/json/single_include/nlohmann/json.hpp:11704:15: warning: unused type alias 'size_type' [-Wunused-local-typedef]
        using size_type = typename BasicJsonType::size_type;
```

I have to block the above warning with the following code:
```cpp
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wunused-local-typedef"
#include <nlohmann/json.hpp>
#pragma clang diagnostic pop
```

--- a/include/nlohmann/detail/json_pointer.hpp
+++ b/include/nlohmann/detail/json_pointer.hpp
@@ -398,7 +398,6 @@ class json_pointer
     */
     BasicJsonType& get_and_create(BasicJsonType& j) const
     {
-        using size_type = typename BasicJsonType::size_type;
         auto result = &j;
 
         // in case no reference tokens exist, return a reference to the JSON value
@@ -471,7 +470,6 @@ class json_pointer
     */
     BasicJsonType& get_unchecked(BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             // convert null values to arrays or objects before continuing
@@ -531,7 +529,6 @@ class json_pointer
     */
     BasicJsonType& get_checked(BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())
@@ -581,7 +578,6 @@ class json_pointer
     */
     const BasicJsonType& get_unchecked(const BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())
@@ -624,7 +620,6 @@ class json_pointer
     */
     const BasicJsonType& get_checked(const BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())
@@ -665,7 +660,6 @@ class json_pointer
     */
     bool contains(const BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())

--- a/single_include/nlohmann/json.hpp
+++ b/single_include/nlohmann/json.hpp
@@ -11529,7 +11529,6 @@ class json_pointer
     */
     BasicJsonType& get_and_create(BasicJsonType& j) const
     {
-        using size_type = typename BasicJsonType::size_type;
         auto result = &j;
 
         // in case no reference tokens exist, return a reference to the JSON value
@@ -11602,7 +11601,6 @@ class json_pointer
     */
     BasicJsonType& get_unchecked(BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             // convert null values to arrays or objects before continuing
@@ -11662,7 +11660,6 @@ class json_pointer
     */
     BasicJsonType& get_checked(BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())
@@ -11712,7 +11709,6 @@ class json_pointer
     */
     const BasicJsonType& get_unchecked(const BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())
@@ -11755,7 +11751,6 @@ class json_pointer
     */
     const BasicJsonType& get_checked(const BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())
@@ -11796,7 +11791,6 @@ class json_pointer
     */
     bool contains(const BasicJsonType* ptr) const
     {
-        using size_type = typename BasicJsonType::size_type;
         for (const auto& reference_token : reference_tokens)
         {
             switch (ptr->type())

